core/src/main/java/org/jdbi/v3/core/mapper/EnumByNameMapperFactory.java

```diff
@@ -17,19 +17,27 @@
 import java.util.Optional;
 
 import org.jdbi.v3.core.config.ConfigRegistry;
+import org.jdbi.v3.core.internal.QualifiedEnumMapperFactory;
 
 import static org.jdbi.v3.core.generic.GenericTypes.getErasedType;
 
 /**
  * Produces enum column mappers, which map enums from varchar columns using {@link Enum#valueOf(Class, String)}.
+ *
+ * @deprecated this class has been superseded by a new implementation
```

Comment:
So people can abandon the deprecated class, sure. But this also raises the question if we shouldn't discourage people from using these at all. They're needed in `resultbearing.mapto` if I'm not mistaken, but outside of that, if these mappers are registered in the chain for example, they'll cause issues.

Reply:
I would point to `Enums.setEnumStrategy(BY_NAME)` as the successor in the deprecated message.

Aside from that, the main reason this class is public was to make it compatible with `@RegisterRowMapperFactory`.

If we add a `@DefaultEnumStrategy(EnumStrategy)` SQL object annotation then we have a viable successor for that scenario as well.

Reply:
> if these mappers are registered in the chain for example, they'll cause issues.

If there is any concrete reason to suspect this will be a problem, we must fix that. We can't deprecate mappers that worked before _and_ break them.